src/commands/calc.rs

```diff
@@ -0,0 +1,74 @@
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::ShellError;
+use nu_protocol::{Primitive, ReturnSuccess, Signature, SyntaxShape, UntaggedValue, Value};
+
+pub struct Calc;
+
+impl WholeStreamCommand for Calc {
+    fn name(&self) -> &str {
+        "calc"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("calc").required(
+            "math-expression",
+            SyntaxShape::String,
+            "the expression to parse into a number",
+        )
+    }
+
+    fn usage(&self) -> &str {
+        "Parse a math expression into a number"
+    }
+
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        calc(args, registry)
+    }
+}
+
+fn calc(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
+    let args = args.evaluate_once(registry)?;
+    let tag = &args.call_info.name_tag;
+    let name_span = tag.span;
+    let input = match args.args.nth(0) {
+        Some(s) => Ok(s),
+        None => Err(ShellError::labeled_error(
+            "Expected a string argument",
+            "requires string input",
+            name_span,
+        )),
+    };
+    let input = input?;
+
+    let output = if let Ok(string) = input.as_string() {
+        match parse(&string, &input.tag) {
+            Ok(value) => ReturnSuccess::value(value),
+            Err(err) => Err(ShellError::labeled_error(
+                format!("Cannot calculate expression: {}", err),
+                format!("error while calculating: {}", err),
+                name_span,
```

Comment:
I think I should change this as it says the same thing 2 times which seems verbose.

screenshot -
![image](https://user-images.githubusercontent.com/32027323/73134594-32b44580-405e-11ea-91de-da6e4bd02c6e.png)

Reply:
If you've got a useful error like that, you can make the message something "Calculator Error" and then the label can be the error message itself.

Reply:
thanks